openapi-core/src/main/java/io/ballerina/openapi/core/generators/client/BallerinaClientGeneratorWithStatusCodeBinding.java

```diff
@@ -222,16 +225,27 @@ private Optional<FunctionDefinitionNode> createImplFunction(String path,
         ImplFunctionSignatureGenerator signatureGenerator = new ImplFunctionSignatureGenerator(operation.getValue(),
                 openAPI, operation.getKey().toString().toLowerCase(Locale.ROOT), path, clientExternFunction);
         //Create function body
-        FunctionBodyGeneratorImp functionBodyGenerator = new ImplFunctionBodyGenerator(path, operation, openAPI,
-                authConfigGeneratorImp, ballerinaUtilGenerator, imports, signatureGenerator.hasHeaders(),
-                signatureGenerator.hasDefaultHeaders(), signatureGenerator.hasQueries(), hasDefaultResponse,
-                nonDefaultStatusCodes);
         FunctionBodyNode functionBodyNode;
-        Optional<FunctionBodyNode> functionBodyNodeResult = functionBodyGenerator.getFunctionBodyNode();
-        if (functionBodyNodeResult.isEmpty()) {
-            return Optional.empty();
+        if (isMock) {
+            MockFunctionBodyGenerator bodyGenerator = new MockFunctionBodyGenerator(path, operation, openAPI,
```

Comment:
The logic after getting the `bodyGenerator` is similar regardless of the mock option. Better if we can extract a method: `getBodyGenerator()` and overwrite it in the `AdvancedMockClientGenerator`. That makes it more cleaner and avoid code duplication

Reply:
Fix with https://github.com/ballerina-platform/openapi-tools/pull/1714/commits/66eb266ab801266ebe424e826da62399c3bbb8ee